components/datetime/src/format/datetime.rs

```diff
@@ -194,23 +214,34 @@ where
         FieldSymbol::Year(year) => match year {
             Year::Calendar => format_number(
                 w,
-                datetime
-                    .datetime()
-                    .year()
-                    .ok_or(Error::MissingInputField)?
-                    .number as isize,
+                fixed_decimal_format,
+                FixedDecimal::from(
+                    datetime
+                        .datetime()
+                        .year()
+                        .ok_or(Error::MissingInputField)?
+                        .number as isize,
+                ),
+                field.length,
+            )?,
+            Year::WeekOf => format_number(
+                w,
+                fixed_decimal_format,
+                FixedDecimal::from(datetime.year_week()?.number as isize),
```

Comment:
Nit, here and above and below and elsewhere:

There should be no reason you would need to cast to `isize`; it looks like `datetime.year_week()?.number` is an `i32`, which is supported in `FixedDecimal::from`.

Reply:
I've fixed a lot of these, but some of the types are not directly convertible, e.g. to construct a `FractionalDecimal` from an `IsoMinute` seems to require first casting to an integer type.